core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3390,6 +3390,11 @@ class KafkaApis(val requestChannel: RequestChannel,
       new DescribeProducersResponse(response.setThrottleTimeMs(requestThrottleMs)))
   }
 
+  def handleUnregisterBrokerRequest(request: RequestChannel.Request): Unit = {
+    throw new UnsupportedVersionException("The broker unregistration API is not available when using " +
+      "Apache ZooKeeper mode.")
```

Comment:
This is a bit confusing since this method can be called when the built-in quorum mode is used too. Or am I missing something?

Reply:
This method can't be called in built-in quorum mode since the API is forwardable and we will always have a forwarding manager

Reply:
Can we add a comment explaining that?

Reply:
will do